crates/duckdb/src/vtab/vector.rs

```diff
@@ -279,4 +295,13 @@ impl StructVector {
         let logical_type = self.logical_type();
         unsafe { duckdb_struct_type_child_count(logical_type.ptr) as usize }
     }
+
```

Comment:
Some of these makes no sense. The struct vector doesn't store the columns directly. Think that goes for all Arrays, Union,

Reply:
Thanks for the comment. This change is about to set the validity on the vector, so duckdb knows when to set null correctly.

See added tests.